pkg/agent/flags/flag_value_sanitizer.go

```diff
@@ -0,0 +1,41 @@
+package flags
+
+// flagValueConstraint represents the constraining limits on a flag value.
+type flagValueConstraint struct {
+	min int64
+	max int64
+}
+
+type flagValueSanitizer struct {
+	constraints map[FlagKey]flagValueConstraint
+}
+
+func NewFlagValueSanitizer(constraints map[FlagKey]flagValueConstraint) *flagValueSanitizer {
+	s := &flagValueSanitizer{
+		constraints: constraints,
+	}
+
+	return s
+}
+
+// Sanitize returns a sanitized (clamped) value for a flag, if constraints for
+// the flag value have been provided. If undefined, the original value is returned.
+func (s *flagValueSanitizer) Sanitize(key FlagKey, value int64) int64 {
```

Comment:
Yeah I thought about string value sanitization. The idea I had was a client provides a sanitizer func like `func (string value) string` which is invoked by the sanitizer code. This would enable each disparate component to specify their own logic for what is, and what is not, a valid string value.

Reply:
I've added a way to easily add sanitization for string flags in the future now.